internal/servers/worker/status.go

```diff
@@ -115,6 +159,33 @@ func (w *Worker) sendWorkerStatus(cancelCtx context.Context) {
 	})
 	if err != nil {
 		w.logger.Error("error making status request to controller", "error", err)
+		// Check for last successful status. Ignore nil last status, this probably
+		// means that we've never connected to a controller, and as such probably
+		// don't have any sessions to worry about anyway.
+		//
+		// If a length of time has passed since we've been able to communicate, we
+		// want to start terminating all sessions as a "break glass" kind of
+		// scenario, as there will be no way we can really tell if these
+		// connections should continue to exist.
+
+		lastStatus := w.LastStatusSuccess()
+		gracePeriod := w.statusGracePeriod()
+		if lastStatus != nil && lastStatus.StatusTime.Before(time.Now().Add(gracePeriod*-1)) {
```

Comment:
Pedantry: the Before/Add stuff could probably be simplified a bit by using `time.Sub`

Reply:
Apparently I thought this didn't exist, because I swear I looked for it 🤦‍♂️ Will fix.

Reply:
Fixed!